react/react1/week3/to-do/src/components/FetchData.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import "react-datepicker/dist/react-datepicker.css";
+import Header from "./Header";
+import TodoItem from "./ToDoItem";
+import DatePicker from "react-datepicker";
+const FetchData = () => {
+  const [todos, setToDos] = useState([]);
+  const [edit, setEEdit] = useState(false);
+  const [newToDo, setNewToDo] = useState({});
+  const [deadLine, setDeadLine] = useState(new Date());
+  const NewToDo = (description) => {
+    setNewToDo({
+      id: Math.floor(Math.random() * (100 - 4 + 1) + 4),
+      description: description,
+      deadline: deadLine,
+      done: false,
+    });
+  };
+  const AddToDo = () => {
+    setToDos((previous) => [...previous, newToDo]);
+  };
+  const DeleteToDo = (id) => {
+    setToDos(todos.filter((eachToDo) => eachToDo.id !== id));
+  };
+  const editMode = (id) => {
+    setEEdit(!edit);
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+  };
+  const handleDesc = (desc, id) => {
+    todos.forEach((todo) =>
+      todo.id === id ? (todo.description = desc) : todo.description
```

Comment:
A simple "if" statement would be better here, as with ternary operator you are returning something for every case that does not match the filtering condition too - that is a lot of processing that does not change anything, but just returns the same todo.description.